src/lib/Microsoft.Health.Fhir.Ingest.Template/TemplateContext.cs

```diff
@@ -0,0 +1,44 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+
+namespace Microsoft.Health.Fhir.Ingest.Template
+{
+    public class TemplateContext<TTemplate> : ITemplateContext<TTemplate>
+        where TTemplate : class
+    {
+        private readonly TTemplate _template;
+        private readonly ICollection<TemplateError> _errors = new Collection<TemplateError>();
+
+        public TemplateContext(TTemplate template)
+        {
+            _template = template;
+        }
+
+        public TTemplate Template
+        {
+            get { return _template; }
+        }
+
+        public ICollection<TemplateError> Errors
```

Comment:
Reason for capturing just the message instead of the full exception?

Reply:
in my understand, the exception conceptually is for wrapping/ending up abnormal situations in one class or one component. However, to the consumer of that class or component, those could be translated into other legal states. For example, the validation logic expects errors to occur. 

I'd be a bit more cautious to use the exceptions for conveying the information, so I'd modeled the message in a meaningful way to the upper level and that model can somehow be shared as well without a dependency on the exception class.  If we need more info, this model can be extended to contain those as well. Hopefully that would make sense.

Reply:
Thanks for the detailed answer.  The part that is confusing to me is the TemplateError isn't for the user of the template.  It is an intermediate that is used to track state that is then in turned used to generate the validation result.  I would be more in favor if it was an interface ITemplateError { string Message {get;} }.  In that case we could have an implementation that wraps an Exception for these exception based errors and later we can add different types when other validations are tracked.

Reply:
Discussed offline, I'll check on this in the next revision.